demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Standard includes. */
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* POSIX socket includes. */
+#include <errno.h>
+#include <netdb.h>
+#include <time.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+/* HTTP API header. */
+#include "http_client.h"
+
+/* Demo config header. */
+#include "demo_config.h"
+
+/**
+ * @brief HTTP server host name.
+ *
+ * This demo uses httpbin.org: A simple HTTP Request & Response Service.
+ */
+#define SERVER_HOST    "httpbin.org"
+
+/**
+ * @brief HTTP server port number.
+ *
+ * In general, port 80 is for plaintext HTTP connections.
+ */
+#define SERVER_PORT    80
+
+/**
+ * @brief Paths for different HTTP methods for specified host.
+ *
+ * For httpbin.org, see http://httpbin.org/#/HTTP_Methods for details on
+ * supported REST API.
+ */
+#ifndef GET_PATH
+    #define GET_PATH    "/get"
+#endif
+
+#ifndef HEAD_PATH
+    #define HEAD_PATH    "/get"
+#endif
+
+#ifndef PUT_PATH
+    #define PUT_PATH    "/put"
+#endif
+
+#ifndef POST_PATH
+    #define POST_PATH    "/post"
+#endif
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS    ( 1000 )
+
+/**
+ * @brief The length in bytes of the user buffer.
+ */
+#define USER_BUFFER_LENGTH                ( 1024 )
+
+/**
+ * @brief Length of an IPv6 address when converted to hex digits.
+ */
+#define IPV6_LENGTH                       ( 40 )
+
+
+/**
+ * @brief Some text to send as the request body for PUT and POST requests in
+ * this demo.
+ */
+#define REQUEST_BODY_TEXT           "Hello, world!"
+#define REQUEST_BODY_TEXT_LENGTH    ( sizeof( REQUEST_BODY_TEXT ) - 1 )
+
+/**
+ * @brief A string to store the resolved IP address from the host name.
+ */
+static char resolvedIpAddr[ IPV6_LENGTH ] = { 0 };
+
+/**
+ * @brief A buffer used to store request headers and reused after sending
+ * the request to store the response headers and body.
```

Comment:
I think this message should be clearer as it is an important point being shown by the demo for space optimization. 
It could be something like
```
/**
 * @brief A buffer used in the demo for storing HTTP request headers and HTTP response headers and body.
 * 
 * @note This demo shows how the same buffer can be re-used for storing the HTTP response after the HTTP request 
 * is sent out. However, the user can also decided to use separate buffers for storing request and response.
 */
```